sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpResponseLogger.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+/**
+ * Manages logging HTTP responses in {@link HttpLoggingPolicy}.
+ */
+public interface HttpResponseLogger {
+    /**
+     * Retrieves the {@link LogLevel} used to log the current response.
+     *
+     * @param defaultLogLevel The default log level to use.
+     * @param response The HTTP response.
+     * @param responseDuration The duration between sending the request and receiving the response.
+     * @return The {@link LogLevel} used to log the current request.
+     */
+    LogLevel getLogLevel(LogLevel defaultLogLevel, HttpResponse response, Duration responseDuration);
+
+    /**
+     * Logs the request.
+     *
+     * @param logger The {@link ClientLogger} used to log the request.
+     * @param logLevel The {@link LogLevel} used to loge the request.
+     * @param response The HTTP response.
+     * @param responseDuration The duration between sending the request and receiving the response.
+     * @return A reactive response that will indicate that the response has been logged.
+     */
+    Mono<HttpResponse> logResponse(ClientLogger logger, LogLevel logLevel, HttpResponse response,
+        Duration responseDuration);
```

Comment:
The `HttpPipelineCallContext` doesn't include the HTTP response. If it did, which seems to have been the intention going by the javadoc on this class, we could have used the `HttpPipelineCallContext` for passing in the response duration as well.

Reply:
Will fix the Javadocs for `HttpPipelineCallContext` as they are misleading.